index.js

```diff
@@ -216,6 +216,20 @@ function resolve(id, opts, cb) {
     opts = opts || {};
     opts.filename = opts.filename || '';
 
+    var cb = function(err, path, pkg) {
```

Comment:
What is this set of nested functions trying to do?

Reply:
1) if it's a module name, resolve to it
2) if it's something in the filesystem, then
  2a) if it's a regular file, resolve to it
  2b) if it's a symlink, resolve to its realpath